lucida/botframework-interface/start_interface.sh

```diff
@@ -0,0 +1,367 @@
+      if [ $? -eq 1 ]; then
+        echo "[ERROR] Please enter a valid host!!!"
+      else
+        break
+      fi
+    done
+    BFW_UPDATE_ENDPOINT=1
+  fi
+  if [ -f "last_pushed_host" ]; then
+    BFW_OLD_HOST=`cat last_pushed_host`
+    if [[ "$BFW_OLD_HOST" != "$BFW_HOST" ]]; then
+      BFW_UPDATE_ENDPOINT=1
+    fi
+  else
+    BFW_UPDATE_ENDPOINT=1
+  fi
+  echo "" >> config.sh
+  echo "# BotFramework bot endpoint" >> config.sh
+  echo "BFW_HOST=\"$BFW_HOST\"" >> config.sh
+
+  if [ -z $BFW_PORT ]; then
+    while [ 1 ]; do
+      echo ""
+      echo "Enter the port on which Lucida BotFramework interface should run."
+      printf "BFW_PORT [3728]: "
+      read BFW_PORT
+      if [ -z $BFW_PORT ]; then BFW_PORT=3728; fi
+      echo $BFW_PORT | grep -Poe "^\d{2,5}$" > /dev/null
+      if [ $? -eq 1 ]; then
+        echo "[ERROR] Please enter a valid port!!!"
+      else
+        break
+      fi
+    done
+  fi
+  echo "" >> config.sh
+  echo "# BotFramework listen port" >> config.sh
+  echo "BFW_PORT=$BFW_PORT" >> config.sh
+
+  if [ -z $BFW_APPID ] || [ -z $BFW_APPPWD ]; then
+    while [ 1 ]; do
+      echo ""
+      echo "Enter Microsoft application ID and password. (View README)"
+      printf "BFW_APPID: "
+      read BFW_APPID
+      echo $BFW_APPID | grep -Poe "^[A-Za-z0-9\-]+$" > /dev/null
+      if [ $? -eq 1 ]; then
+        echo "[ERROR] Please enter a valid application ID!!!"
+        continue
+      fi
+      printf "BFW_APPPWD: "
+      read BFW_APPPWD
+      echo $BFW_APPPWD | grep -Poe "^[A-Za-z0-9]+$" > /dev/null
+      if [ $? -eq 1 ]; then
+        echo "[ERROR] Please enter a valid application password!!!"
+      else
+        break
+      fi
+    done
+  fi
+  echo "" >> config.sh
+  echo "# BotFramework credentials" >> config.sh
+  echo "BFW_APPID=\"$BFW_APPID\"" >> config.sh
+  echo "BFW_APPPWD=\"$BFW_APPPWD\"" >> config.sh
+
+  if [[ "$BFW_HOST" == "https://xxx.ngrok.io" ]] && [ $BFW_SKIP_UPDATE -eq 0 ]; then
+    USE_NGROK=1
+    BFW_HOST=
+    ps -ef | grep ngrok | grep http | grep $BFW_PORT > /dev/null
+    if [ $? -eq 1 ]; then
+      ngrok http $BFW_PORT > /dev/null & disown
+      echo ""
+      echo "[INFO] Waiting for ngrok to go online..."
+      sleep 10
+      ps -ef | grep ngrok | grep http | grep $BFW_PORT > /dev/null
+      if [ $? -eq 1 ]; then
+        echo "[ERROR] Could not start ngrok client!!! Please fix the issue"
+        exit 1
+      fi
+    fi
+    while read line; do
+      ngrok_pid=`echo $line | grep -Poe "\d+(?=/ngrok)"`
+      ngrok_port=`echo $line | grep -Poe "(?<=127.0.0.1:)\d+(?=[^\d])"`
+      if [ ! -z $ngrok_pid ] && [ ! -z $ngrok_port ]; then
+        ps -ef | grep $ngrok_pid | grep $BFW_PORT > /dev/null
+        if [ $? -eq 0 ]; then
+          BFW_HOST=`curl -s localhost:$ngrok_port/inspect/http | grep $BFW_PORT | grep -Poe "https://.*?ngrok.io"`
+        fi
+      fi
+    done < <(netstat -peanut 2>/dev/null | grep LISTEN)
+    if [ -z $BFW_HOST ]; then
```

Comment:
In my case on my local machine, I get an error `start_interface.sh: line 278 [: too many arguments`. When asking `if[ -z $___ ]` we need parenthesis around the variable like `if[ -z "$___" ]` because the variable could have spaces. Can you change all of the string checks to have ""?